pkg/prometheus/promcfg_test.go

```diff
@@ -4358,6 +4367,7 @@ func TestBodySizeLimits(t *testing.T) {
 							Interval: "30s",
 						},
 					},
+					BodySizeLimit: tc.serviceMonitorBodySizeLimit,
```

Comment:
```suggestion
					BodySizeLimit: ptr.To(tc.serviceMonitorBodySizeLimit),
```
You can create a pointer like this, then no need to use pointers before this line.
That way you can also delete the line `expectedServiceMonitorBodySizeLimit := monitoringv1.ByteSize("2GB")`